utils_cv/detection/dataset.py

```diff
@@ -107,10 +150,44 @@ def parse_pascal_voc_anno(
             os.path.join(anno_dir, root.find("filename").text)
         )
 
-    # extract bounding boxes and classification
+    # extract bounding boxes, classification and keypoints
     objs = root.findall("object")
     for obj in objs:
         label = obj.find("name").text
+        # Get keypoints if any
+        if keypoint_meta is not None:
+            kps = []
+            kps_category = keypoint_meta["category"]
+            kps_labels = keypoint_meta["labels"]
+
+            # It seems Keypoint R-CNN model doesn't accept varied number of
+            # keypoints, so we only allow one category currently.
+            if label != kps_category:
```

Comment:
Not sure we want to implement this. Seems odd to have a dataset and then only read some of the images. The likely better alternative is to create a subset of the dataset with just e.g. milk bottle images. See also my comment in the notebook. Furthermore, having multiple objects I believe is ok IF they have the same number of keypoints.

Reply:
Generated a dataset with milk bottle.  Yes, it is OK to have multiple objects of the same category in a image.